### PR TITLE
API for getting all client items

### PR DESCRIPTION
**Pull request by CrazyHackGUT**
This PR implements new API method: `Shop_GetClientItems()`. This method accepts client index on server, and returns the array with item ids which player have.

Method documented in include-file.

### CODE REVIEW

File: addons/sourcemod/scripting/include/shop/players.inc [hunk 1]
```diff
@@ -250,4 +250,15 @@ native void Shop_SetClientItemCount(int client, ItemId item_id, int count = 0);
  *	-
  *	@return true on success and false otherwise
 */
-native bool Shop_UseClientItem(int client, ItemId item_id);
\ No newline at end of file
+native bool Shop_UseClientItem(int client, ItemId item_id);
+
+/**
+ * Returns the array with player item ids.
```
Comment (R1KO) [inline addons/sourcemod/scripting/include/shop/players.inc]: may be add `Must be freed with CloseHandle()/delete`

File: addons/sourcemod/scripting/shop/player_manager.sp [hunk 2]
```diff
@@ -635,6 +636,32 @@ public int PlayerManager_ToggleClientCategoryOff(Handle plugin, int numParams)
...
+public int PlayerManager_GetClientItems(Handle plugin, int numParams)
+{
+	int client = GetNativeCell(1);
+	
+	char error[64];
+	if (!CheckClient(client, error, sizeof(error)))
+		ThrowNativeError(SP_ERROR_NATIVE, error);
+
+	ArrayList hArrayItems = new ArrayList(ByteCountToCells(4));
+	KeyValues hKVItems = h_KvClientItems[client];
```
Comment (R1KO) [inline addons/sourcemod/scripting/shop/player_manager.sp]: it's just for convenience, you exactly don't need to close it ?

Comment (CrazyHackGUT) [reply addons/sourcemod/scripting/shop/player_manager.sp]: Close what? `hKVItems`? I just copy pointer for easy reading.
If i close `hKVItems`, `h_KvClientItems[client]` will be closed too.

**Comment by CrazyHackGUT** [comment]
> each time you will use this function, it may cause shop to work heavily

We can leave this comment in notes in include.

Also, need test this native. I don't tested yet.
I wrote simple plugin for test. [Download here](https://github.com/FD-Forks/Shop-Core/files/3349787/ShopPIMTest.zip). Adds command `sm_shop_dumpmyitems` for this purpose.

**Pull request merged**